L1Trigger/VertexFinder/interface/VertexProducer.h

```diff
@@ -0,0 +1,49 @@
+#ifndef __L1Trigger_VertexFinder_VertexProducer_h__
+#define __L1Trigger_VertexFinder_VertexProducer_h__
+
+#include "DataFormats/L1TrackTrigger/interface/TTTypes.h"
+#include "DataFormats/L1Trigger/interface/Vertex.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "L1Trigger/VertexFinder/interface/AlgoSettings.h"
+#include "L1Trigger/VertexFinder/interface/RecoVertex.h"
+#include "L1Trigger/VertexFinder/interface/VertexFinder.h"
+
+#include <iostream>
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+
+namespace l1tVertexFinder {
+  class AlgoSettings;
+}
+
+class VertexProducer : public edm::EDProducer {
```

Comment:
Please use a thread-efficient base class
https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkModuleTypes

Reply:
Please also move the contents of this header to the corresponding source file (headers for modules are largely unnecessary).